rust/candid/tests/parse_value.rs

```diff
@@ -222,3 +227,29 @@ fn parse_shorthand() {
     let args = parse_args("(variant { 0x2a }, variant { label })");
     assert_eq!(format!("{}", args), "(variant { 42 }, variant { label })");
 }
+
+#[test]
+fn parse_annval() {
+    let args = parse_args("((1))");
+    assert_eq!(
+        args,
+        IDLArgs {
+            args: vec![IDLValue::Number("1".to_string())]
+        }
+    );
+
+    let args =
+        parse_args("((1 : nat), (123 : int), (456 : int32), (789 : float32), (1011 : float64))");
```

Comment:
This is what we should be able to parse

```suggestion
        parse_args("(1 : nat, (123 : int), 456 : int32, 789. : float32, 1011.0 : float64)");
```